src/main/java/net/snowflake/client/jdbc/SnowflakeSQLLoggedException.java

```diff
@@ -227,7 +255,7 @@ public SnowflakeSQLLoggedException(
     // add telemetry
     String reason =
         errorResourceBundleManager.getLocalizedMessage(String.valueOf(vendorCode), params);
-    sendTelemetryData(null, reason, SQLState, vendorCode, null, session);
+    sendTelemetryData(null, reason, SQLState, vendorCode, null, session, this);
```

Comment:
nit; the parameters of `sendTemetryData` is cluttered. You should consider changing this to a builder pattern for better readability of code.

Reply:
There are a lot of parameters. But my hesitation to changing to a builder pattern is that since this function is static and used in 2 entirely separate classes, using the builder would require creating yet another class (SQLExceptionTelemetry or something) with another builder class inside it (SQLExceptionTelemetryBuilder) to encapsulate all of the parameters, and each instance of the Sqlexception would have to instantiate an instance of this new class first before calling **sendTelemetryData**. I think this might make the code less readable; it would work better if this function was only being used in 1 class.